DataFormats/Detectors/CTP/src/ctpRateFetcher.cxx

```diff
@@ -0,0 +1,182 @@
+{
+  setupRun(runNumber, ccdb, timeStamp);
+  if (sourceName.find("ZNC") != std::string::npos) {
+    if (runNumber < 544448) {
+      return fetchCTPratesInputs(ccdb, timeStamp, runNumber, 25) / (sourceName.find("hadronic") != std::string::npos ? 28. : 1.);
+    } else {
+      return fetchCTPratesClasses(ccdb, timeStamp, runNumber, "C1ZNC-B-NOPF-CRU", 6) / (sourceName.find("hadronic") != std::string::npos ? 28. : 1.);
+    }
+  } else if (sourceName == "T0CE") {
+    return fetchCTPratesClasses(ccdb, timeStamp, runNumber, "CMTVXTCE-B-NOPF");
+  } else if (sourceName == "T0SC") {
+    return fetchCTPratesClasses(ccdb, timeStamp, runNumber, "CMTVXTSC-B-NOPF");
+  } else if (sourceName == "T0VTX") {
+    if (runNumber < 534202) {
+      return fetchCTPratesClasses(ccdb, timeStamp, runNumber, "minbias_TVX_L0", 3); // 2022
+    } else {
+      double_t ret = fetchCTPratesClasses(ccdb, timeStamp, runNumber, "CMTVX-B-NOPF");
+      if (ret < 0.) {
+        LOG(info) << "Trying different class";
+        ret = fetchCTPratesClasses(ccdb, timeStamp, runNumber, "CMTVX-NONE");
+        if (ret < 0) {
+          LOG(fatal) << "None of the classes used for lumi found";
+        }
+      }
+      return ret;
+    }
+  }
+  LOG(error) << "CTP rate for " << sourceName << " not available";
+  return -1.;
+}
+double ctpRateFetcher::fetchNoPuCorr(o2::ccdb::BasicCCDBManager* ccdb, uint64_t timeStamp, int runNumber, std::string sourceName)
+{
+  setupRun(runNumber, ccdb, timeStamp);
+  if (sourceName.find("ZNC") != std::string::npos) {
+    if (runNumber < 544448) {
+      return fetchCTPratesInputsNoPuCorr(ccdb, timeStamp, runNumber, 25) / (sourceName.find("hadronic") != std::string::npos ? 28. : 1.);
+    } else {
+      return fetchCTPratesClassesNoPuCorr(ccdb, timeStamp, runNumber, "C1ZNC-B-NOPF-CRU", 6) / (sourceName.find("hadronic") != std::string::npos ? 28. : 1.);
+    }
+  } else if (sourceName == "T0CE") {
+    return fetchCTPratesClassesNoPuCorr(ccdb, timeStamp, runNumber, "CMTVXTCE-B-NOPF");
+  } else if (sourceName == "T0SC") {
+    return fetchCTPratesClassesNoPuCorr(ccdb, timeStamp, runNumber, "CMTVXTSC-B-NOPF");
+  } else if (sourceName == "T0VTX") {
+    if (runNumber < 534202) {
+      return fetchCTPratesClassesNoPuCorr(ccdb, timeStamp, runNumber, "minbias_TVX_L0", 3); // 2022
+    } else {
+      double_t ret = fetchCTPratesClassesNoPuCorr(ccdb, timeStamp, runNumber, "CMTVX-B-NOPF");
+      if (ret < 0.) {
+        LOG(info) << "Trying different class";
+        ret = fetchCTPratesClassesNoPuCorr(ccdb, timeStamp, runNumber, "CMTVX-NONE");
+        if (ret < 0) {
+          LOG(fatal) << "None of the classes used for lumi found";
+        }
+      }
+      return ret;
+    }
+  }
+  LOG(error) << "CTP rate for " << sourceName << " not available";
+  return -1.;
+}
+double ctpRateFetcher::fetchCTPratesClasses(o2::ccdb::BasicCCDBManager* /*ccdb*/, uint64_t timeStamp, int /*runNumber*/, const std::string& className, int inputType)
```

Comment:
there can be parallel runs and so two objecys compatible with timestamp

Reply:
But in this function the run number is not used at all, since the `mScalers` object is already initialized. The run  number is only used by `fetch()` and `fetchNoPuCorr()`, where you call `setupRun()` at the beginning.

Reply:
removed (probably development relict)